### ISSUE TITLE
Legacy version ReactiveCocoa's API

### ISSUE THREAD

**Issue by EvilNOP** [ReactiveCocoa/ReactiveCocoa#3327]
Hey, with the following code:
```swift
let signal = self.rac_signalForSelector(Selector(...), fromProtocol: protocol)

signal.filterAs { (tuple: RACTuple) -> Bool in }
```

Is there a similar code corresponding to the current ReactiveCocoa's version(5.0.0-alpha.3)?

**Comment by andersio** [comment]
`signal(for:)` in #3302 seems like what you need. But it might take a while to land since we are busying in getting ReactiveSwift into stable release.

```
self.reactive.signal(for: #selector(...)).map { $0[0] as! Bool }.filter { $0 }
```

However, we planned to bring the ability to dynamically implement ObjC protocol requirements over in this form. Before we eventually roll out a refined solution, you will have to implement an empty body in your implementation.

**Comment by PrekshyaBasnet** [comment]
@EvilNOP 
```Swift
let keywords = self.reactive.signal(for: #selector(UISearchBarDelegate.searchBar(_:textDidChange:)))
      .map({ (text) -> String in
        //first in array is SearchBar and second is SearchText as mentioned in protocol
        return text[1] as! String 
      })
extension ViewController: UISearchBarDelegate {
//had to add this delegate or the app crashes
 func searchBar(_ searchBar: UISearchBar, textDidChange searchText: String) {    
  }
}
```

**Comment by EvilNOP** [comment]
@PrekshyaBasnet My point is straight forward, that's in the condition of not conforming a protocol, I can still get its signal.

Your method is just what andersio said, I already know it, but thx again.

As andersio said that the what I need might take a while to land.

**Comment by andersio** [comment]
Resolved by #3302.
